test/run-jest.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+set -x
+set -e
+set -u
+set -o pipefail
+
+# For storing PIDs of subshells
+PIDS=(0 0)
+
+yarn 'jest' '--config=./jest.config.js' "$@" & PIDS[0]=$!
```

Comment:
This pattern seems questionable :thinking: This means if one process fails, the other will keep going and there will be no way to stop it. Sounds like a pain.

We have a package that is supposed to do this sort of thing for us: `concurrently`. I don't think it works correctly in all cases, but it might work more often than this.

I think we could scrap this whole script and call `concurrently` directly from the npm script.

Reply:
> This means if one process fails, the other will keep going and there will be no way to stop it. Sounds like a pain.

This is an intentional feature, not a bug! The idea being that, in CI, you want to know of all unit test failures, not just the ones for a particular config. This is an attempt to replicate what the Jest `projects` feature does, which is to run all tests for every project and then print the complete results.

If developing locally, the tests for a particular config can be run by doing `yarn jest --config=PATH_TO_CONFIG`.

On the topic of using `concurrently`, that seems like a good idea, since it appears to be a well-maintained tool that does exactly what I'm trying to do. We may still have to keep this script in order to do something like `yarn test:unit:jest --watch` and pass the `--watch` argument to both invocations of `jest`. That's what the `"$@"` is about in the Bash script.

Reply:
Just to be clear, when I say "there is no way to stop it", what I mean is that the interrupt signal no longer halts the script. I certainly hope disabling CTRL+C wasn't intentional! It's incredibly frustrating when scripts can't be halted.

Reply:
... I tested this, and it works :thinking: I guess it's because these tests don't spawn child processes? This was a struggle to get working with the e2e tests.. Our attempts to combine tests just with bash resulted in them continuing on after interrupting the main process.

Reply:
After some deliberation, I decided to go with `concurrently` but keep the bash script. I went for `concurrently` because a tool we're already using is probably better than a weird Bash loop-and-`wait` implementation. However, I invoke `concurrently` in the `.sh` file to preserve the ability to pass arguments to both `jest` processes by doing e.g. `yarn test:unit:jest --silent`.

662f5ae